persist/fs/commitlog/iterator.go

```diff
@@ -95,6 +95,7 @@ func (i *iterator) Next() bool {
 	var err error
 	i.read.series, i.read.datapoint, i.read.unit, i.read.annotation, err = i.reader.Read()
 	if err == io.EOF {
+		i.reader.Close()
 		// Try the next reader
 		i.reader = nil
```

Comment:
We should probably add a method that encapsulates `i.reader = nil; i.reader.Close()` as it's two steps performed often in this file.